File: 1051/main.rb
# @param {Integer[]} heights
# @return {Integer}
def height_checker(heights)
    n = heights.size
    rightPosition = heights.sort
    
    result = 0
    n.times do |i|
        result += 1 if rightPosition[i] != heights[i]
    end
    
    result
end

File: 1051/main.java
class Solution {
    public int heightChecker(int[] heights) {
        int n = heights.length;
        int[] rightPosition = new int[n];
        for(int i = 0; i < n; i++) {
            rightPosition[i] = heights[i];
        }
        
        Arrays.sort(rightPosition);
        
        int result = 0;
        for(int i = 0; i < n; i++) {
            if(rightPosition[i] != heights[i]) {
                result++;
            }
        }
        
        return result;
    }
}
